Optional interactive interpreters

Mytho opened the pull request:
Implements issue #210.

This allows the optional use of bpython or IPython as the interactive interpreter. I used the referenced [Django code](https://github.com/django/django/blob/9996158db467f9bef8243fcc36dc3602570e3613/django/core/management/commands/shell.py) as an example for this implementation.

davidszotten commented:
hey, thanks for the contribution! will try to find time to review as soon as possible

davidszotten commented on nameko/cli/shell.py:
```diff
@@ -13,10 +13,19 @@
+    parser.add_argument(
+        '--interface', choices=SHELLS,
+        help='Specify an interactive interpreter interface.')
+    parser.add_argument(
```
don't think we need a separate option for this (think django has this for historical reasons. tried digging through the source and got as far as the svn->git conversion and lost track. from what i can tell, django originally had `--plain` to stop defaulting to ipython, and only later added `--interface` to support bpython

here i guess we could just add 'plain' as another choices in `SHELLS`?

davidszotten commented on nameko/cli/shell.py:
```diff
@@ -42,6 +51,27 @@ def make_nameko_helper(config):
+
+    for shell in available_shells:
+        try:
+            return globals()[shell](banner, local)
```
not so keen on using `globals` like this

davidszotten commented on nameko/cli/shell.py:
```diff
@@ -53,11 +83,17 @@ def main(args):
+
+        run_shell(shell=args.interface, banner=banner, local=ctx)
+    except ImportError:
+        # Support the regular Python interpreter startup script if someone
```
from what i can tell (using latest versions on pypi, might differ between versions), calling `embed` in bpython causes pythonstartup to be executed, but in ipython. one option is to run it "manually" for all shells (like this), and then unset the environment variable to prevent it from running again. thoughts?

Mytho commented:
Thanks for your constructive feedback.

I have:
- removed the `--plain` argument and made `plain` an available shell.
- added `ShellRunner` responsible for starting the interactive shells, so the globals are gone.
- added the manual startup script before each of the interactive shells.

davidszotten commented:
cool. had a quick look at drying up the tests and noticed they fail if bpython or ipyhton are installed. added a fix on top (see separate pr)
